### ISSUE TITLE
Display of attribute pricing: use of "+"

### ISSUE THREAD

**Issue by torvista** [zencart/zencart#6084]
Product
Base price: 100
Attribute price: 20
Tax: 21% => total price 145.20.

It appears the display of the '+' is not based on logic, but on whether the '+' was selected with the attribute definition.

Priced by Attributes
Has the '+' which is misleading,
![2023-12-13 13_14_51-Test attributes WH - €145 20 _ Zen Cart!, The Art of E-commerce — Mozilla Firefo](https://github.com/zencart/zencart/assets/4391026/816a1eb6-5931-4489-a92b-5617b6d1153a)

Not Priced by Attributes (ok):
![2023-12-13 13_14_38-Test attributes WH - €121 00 _ Zen Cart!, The Art of E-commerce — Mozilla Firefo](https://github.com/zencart/zencart/assets/4391026/efbf7341-ba16-4880-9759-dda5c06f168a)

If the '+' is removed from the attribute, it does not display. Which is misleading, it **should** display (yes I changed sold out-add to cart button option meanwhile).
![image](https://github.com/zencart/zencart/assets/4391026/55b36a07-a826-47d0-8666-c27fb3176316)

**Comment by mc12345678** [comment]
Why would a product that has only one attribute have an attribute price other than zero with a non zero base price? 

The expectation of setup for the presented product would be that the base price is 120, the attribute would have no price entered. 

There may be a case, but as shown yes this is confusing.

**Comment by lat9** [comment]
> Why would a product that has only one attribute have an attribute price other than zero with a non zero base price?
> 
> The expectation of setup for the presented product would be that the base price is 120, the attribute would have no price entered.
> 
> There may be a case, but as shown yes this is confusing.

Agreed for the radio (or dropdown) attribute type, but I can see a _single_ optional checkbox/file/text attribute adding to a product's price if selected.

**Comment by mc12345678** [comment]
While there is varying assistance in the docs, this one at least partially suggests what is observed: https://docs.zen-cart.com/user/admin_pages/catalog/attributes_controller/#weights-and-prices
While this one tends to lead to the confusion somewhat described (e.g. not helpful in regards to this discussion but offers assistance on other attribute discussion): https://docs.zen-cart.com/user/products/attribute_pricing/#product-priced-by-attributes-vs-attributes-with-prices

Based on information such as that (including this ancient thread: https://www.zen-cart.com/showthread.php?4810-Attributes-HOW-TO&p=33097#post33097) has been what is the impact of the attribute's number on the lowest price of the product where the lowest attribute product price should not add to the base price of the product. It should either be the lowest price of the product or should be zero. 

Unfortunately, in the realm of eliminating attributes from display because of associated availability, subsequent price display isn't as easily coordinated in setup and likely require some form of logic processing to support that display functionality.

**Comment by lat9** [comment]
Is this, possibly, corrected by these PRs: https://github.com/zencart/zencart/pull/6647 and https://github.com/zencart/zencart/pull/6565?

**Comment by torvista** [comment]
...tested with todays Github.

To refresh, 
make a world zone of 21% tax, 
make a product with a base price of 100 (=> 121 with tax).
create an attribute with a price of 16.5289
display prices with tax to true

priced by attributes NO:
![priced by attributes NO](https://github.com/user-attachments/assets/5c341dfd-26d2-41a6-ae79-3ada5d2cf923)

priced by attributes YES:

![priced by attributes YES](https://github.com/user-attachments/assets/be7827c8-7b42-4bbe-89c4-729697bf960a)

Yes they look the same.

The + is only shown if manually selected in attribute controller. I don't think there is a way to display it automatically, as it depends on how you are choosing to use these options.

Removing the NONE attribute

priced by attributes NO:

![priced by attributes NO no NONE](https://github.com/user-attachments/assets/bfcbed7f-9115-4823-a307-54394d54cb9d)

priced by attributes YES:
![priced by attributes YES no NONE](https://github.com/user-attachments/assets/5894ac91-109a-417d-8016-9e8cd3c6e4f6)

I have always done the following:
- if all attributes are the same price: set one price in Edit Product / attributes prices all zero / priced by attributes NO
- if any attributes have different prices: set price in Edit Product to zero / set each attribute to full price / priced by attributes YES

From the docs

> When Product Priced by Attributes is set to Yes, the product prices include the attribute price. When it is set to no, the prices are shown as increments on top of the product price.
![image](https://github.com/user-attachments/assets/58cc1594-bea0-4adc-ae00-a0b3828eaed0)

This implies this visual difference is ONLY due to changing Priced by Attributes...true?

I think If using incrementally-priced attributes, you should use a base price with a no-price attribute for that base price.

In conclusion I think it is correct, but perhaps the docs need more detail and examples, from other viewpoints.
